src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/MessageProducerImplUnitTest.java

```diff
@@ -48,22 +54,27 @@
     @InjectMocks
     private MessageProducerImpl messageProducerImpl;
 
+    @Captor
+    private ArgumentCaptor<Message> kafkaMessageCaptor;
+
     @Test
     void testSuccessfulWriteToTopic()
             throws ServiceException, ExecutionException, InterruptedException {
         when(producer.sendAndReturnFuture(any())).thenReturn(mockedFuture);
 
-        messageProducerImpl.writeToTopic(getDummyChipsRestInterfacesSend());
-
+        messageProducerImpl.writeToTopic(getDummyChipsRestInterfacesSend(), TEST_TOPIC);
         verify(mockedFuture, times(1)).get();
+        verify(producer, times(1)).sendAndReturnFuture(kafkaMessageCaptor.capture());
+        Message kafkaMessage = kafkaMessageCaptor.getValue();
+        assertEquals(kafkaMessage.getTopic(), TEST_TOPIC);
```

Comment:
expected and actual arguments are the wrong way around

Reply:
Picked it up cna changed it.